### PR TITLE
Upgrade kubernetes version to 1.8.4

### PR DESCRIPTION
**Pull request by pipo02mix**

### CODE REVIEW

**Comment by pipo02mix** [comment]
I have tested the new k8s version (1.8.4). First manually and then I have run the conformance tests.

```
<testsuite tests="125" failures="0" time="2732.256172625">
```

Do I need to change glide.yaml from sha version to master before the merge?

cc @puja108

**Comment by puja108** [comment]
yeah, I you need to revert you glide.yaml change to master and then vendor

**Comment by pipo02mix** [comment]
ready to be merged 

cc @puja108 @fgimenez @rossf7 @jgsqware

**Comment by rossf7** [comment]
@pipo02mix In your branch I'm still seeing 1.8.1 in k8scloudconfig in the vendor dir. Can you double check?

https://github.com/giantswarm/aws-operator/blob/8b67fbe8c3023ec19a5d4c5c9c8dd70f228bce39/vendor/github.com/giantswarm/k8scloudconfig/v_0_1_0/master_template.go

There are also some new versions added to k8scloudconfig by @xh3b4sd. Please make sure your changes are in all versions.

See https://github.com/giantswarm/k8scloudconfig/pull/260/files

**Comment by pipo02mix** [comment]
Wow then glide has not update from master because I see my changes on k8scloudconfig.

**Comment by oponder** [comment]
BTW: This should've incremented the version bundle. So for future reference, we should think about what kind of changes are being introduced in a PR and if it means that there is a new version bundle to be made.

To be honest, I don't understand this process yet.

What happens when a new version bundle is made? In Happa people are able to select different releases versions. How does AWS-Operator deal with a request for a v0.3.0 vs a v0.2.0 cluster. 

In other words, are we having some `if` statements here and there that select what version of k8scloudconfig will be used? How is this supposed to work. I don't get it yet.

**Comment by oponder** [comment]
@pipo02mix not saying you did something wrong btw. The message above goes towards the reviewers as well. This Release Version is a new concept and we're not used yet to thinking about consequences of a PR and if that means a new Release Version needs to be made.

**Comment by xh3b4sd** [comment]
@oponder the conditionals are inside the resource router. See https://github.com/giantswarm/aws-operator/blob/master/service/resource_router.go.

**Comment by oponder** [comment]
To further sharpen my question, now that we  have a `1.8.4 k8s`, is it possible to select a previous version and get a `1.8.1 cluster`? If so let me know. 

My understanding right now is that this isn't possible, so the previous versions need to be deprecated.

**Comment by xh3b4sd** [comment]
> is it possible to select a previous version and get a `1.8.1 cluster`? If so let me know.

No we use the same cloudconfig for all resources. Such kind of version has still to be implemented. The operator is not yet prepared to work like this. You are right, the former version bundles have to be deprecated.

**Comment by pipo02mix** [comment]
@oponder no worries, I did something wrong but I did not be aware of the consequences. :)

Now I have clear what is the purpose of a release and how operators expose version bundles.
